client/src/component/Addvideo.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+
+function AddVideo({ videoData }) {
+  const [showForm, SetShowForm] = useState(false);
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+  const [allData, setAllData] = useState(videoData);
+
+  const handleShowForm = () => {
+    SetShowForm(!showForm);
+  };
+  const handleAddVideo = (event) => {
+    event.preventDefault();
+
+    const newVideoData = {
+      id: Math.floor(Math.random() * 1000),
+      title,
+      url,
+    };
+
+    fetch(videoData, {
```

Comment:
We are doing a `fetch` here, but we are doing nothing with the response. The code is also not asynchronous by using async/await or Promise chaining (.then().then().catch().finally())

If the fetch fails or is successful, we have no way of knowing. I would suggest here we use async/await and save the response to a const. With that, we can let the user know whether they added the video successfully, or if it failed.

We should also try to introduce try/catch here too.

So for example
```
try
{
   const res = fetch(....

    if(res.status !== 200)
        throw new Error("Video upload failed!")

     setAllData({ ...allData, ...form})
     setForm({
        id: "",
        title: "",
        ...
{
catch(error)
{
  console.log("error: ", error)
}
```

I've not written this out completely, but just as guidance on how I would do this. Again, should this not make sense we can discuss over a video call again soon :)